src/realm/object-store/dictionary.hpp

```diff
@@ -34,18 +35,95 @@ class Dictionary : public object_store::Collection {
 
     template <typename T>
     void insert(StringData key, T value);
+    template <typename T>
+    T get(StringData key) const;
+
+
+    template <typename T, typename U, typename Context>
+    void insert(Context&, T&& key, U&& value, CreatePolicy = CreatePolicy::SetLink);
+    template <typename Context, typename T>
+    auto get(Context&, T key) const;
 
 private:
     realm::Dictionary* m_dict;
+
+    template <typename Fn>
+    auto dispatch(Fn&&) const;
+    Obj get_object(StringData key) const;
 };
 
+
+template <typename Fn>
+auto Dictionary::dispatch(Fn&& fn) const
+{
+    verify_attached();
+    using PT = PropertyType;
+    auto type = get_type();
+    switch (type & ~PropertyType::Flags) {
```

Comment:
There shouldn't be any need to repeat this switch here rather than using switch_on_type

Reply:
I originally used `switch_on_type`, but that introduced some linking problems as it required some operations on Mixed, that we currently don't have. This can be seen as a temporary fix.

Reply:
Would be good with a FIXME about that